tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/RawTensor.java

```diff
@@ -219,9 +221,16 @@ private static int dtype(TF_Tensor handle) {
   RawTensor(TensorTypeInfo<? extends TType> typeInfo, Shape shape) {
     this.typeInfo = typeInfo;
     this.shape = shape;
+
+    TensorScope currentScope = TensorScope.currentScope();
+    if (currentScope != null) {
+      this.scope = currentScope.withAttached(this);
+    }
   }
 
   private PointerScope tensorScope;
+  private boolean closed;
```

Comment:
Instead of carrying this extra boolean to check if a tensor has been closed, I think we can rely on `tensorHandle.isNull()` or something like that.

Reply:
The reason that I didn't is if we somehow have multiple `RawTensor`s with the same native handle, the handle will be reference counted by PointerScope, so you could close a tensor but the handle would still be open b/c another tensor has it.

Reply:
If more than one `RawTensor` instance carries the same pointer to a native tensor, then something else is going really wrong since they have a one-to-one relationship. Therefore I would assume it's always the case and we can get rid of this boolean.

Reply:
@karllessard is right. However, I suppose it's possible for TF Core to return an existing buffer that was passed as input, but in that case, the wrapping TF_Tensor would also not be the same, and tracking it at this level with booleans and what not, without comparing the actual pointer values, isn't going to work. @rnett To do something about this we'd need to compare pointer addresses somehow, but we should probably first make sure that there are cases when TF Core returns the same buffer multiple times like that.

Reply:
The only operations that tensors are usually used with anyways are `constantOf` and `feed`, I think using `isNull()` is good enough.  It never gives false positives, the worst that can happen is letting you access a tensor that you "shouldn't" be able to.